spec/counter_culture_spec.rb

```diff
@@ -1711,6 +1712,24 @@ def yaml_load(yaml)
     ConditionalMain.find_each { |main| expect(main.conditional_dependents_count).to eq(main.id % 2 == 0 ? 3 : 0) }
   end
 
+  it "should correctly fix the counter caches for thousands of records when counter is conditional using :if/:unless" do
```

Comment:
We'd need a test for the normal updating on insert / update / delete as well, right? I don't even think we change `fix_counts` in this PR, that's just based on `column_names` so I don't even think this test here adds any test coverage we don't already have.

Reply:
Yeah I agree with sidestepping the fix counts logic since that's orthogonal. I tried to think of a way to have fix_count work well out of the box with this approach, but it would require counting in ruby instead of sql.